### PR TITLE
Add-Modal-Dialog-Razor-Template

### PR DESCRIPTION
**Pull request by ArielBlanco1990**
## Description

This update contains a new OpenContent template "Modal-Dialog". This template has the functionality to launch a modal to the user when entering the page or when called from a button.

## How Has This Been Tested?

DNN 9.13.6

## Screenshots (if appropriate):
image
image
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: Modal-Dialog/Template.cshtml [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+          </div>
+        </div>
+        <div class="row col-12 modalDialog-container">
+          <div class="col-12 mt-15">
+            <p class="modalDialog-content">
+              @modalContent
+            </p>
+            @if(IsAuthenticated){
+            <p class="modalDialog-content">
+              <strong>Current email:</strong> @userEmail
```
Comment (WillStrohl) [inline Modal-Dialog/Template.cshtml:143]: @ArielBlanco1990 Same localization question.

File: Modal-Dialog/Template.cshtml [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+  var modalId = Model.ModalId;
+  var modalDialogTitle = Model.ModalDialogTitle;
+  var modalContent = Model.ModalContent;
+  var modalHTMLContent = Model.HTMLContent;
+  var primaryButtonText = Model.PrimaryButtonText;
+  var primaryButtonLink = Model.PrimaryButtonLink;
+  var secondaryButtonText = Model.SecondaryButtonText;
+  var secondaryButtonLink = Model.SecondaryButtonLink;
+  bool IsAuthenticated = User.Identity.IsAuthenticated;
+  var userEmail = User.Identity.IsAuthenticated ? User.Identity.Name : "No user logged in";
```
[the remark this answers is not in the document]
Comment (ArielBlanco1990) [reply Modal-Dialog/Template.cshtml:120]: @WillStrohl I'll fix it in a moment.

File: Modal-Dialog/Template.cshtml (hunk 1 above)
Comment (ArielBlanco1990) [reply Modal-Dialog/Template.cshtml:143]: @WillStrohl I'll fix it in a moment.

**Comment by ArielBlanco1990** [comment]
@WillStrohl I have updated the template, now the texts are obtained from the module settings, if they are empty they will not be displayed.

**Review by WillStrohl — APPROVED** [review]
Looks good!

**Pull request merged**